### ISSUE TITLE
Login attempt or request with invalid authentication

### ISSUE THREAD

**Issue by Nezz** [home-assistant/iOS#3138]
**iOS device model, version and app version**

Model Name: iPhone 15 Pro
Software Version: iOS 18.1
App version: 2024.934

**Home Assistant Core Version**
2024.11.1

**Describe the bug**
Every now and then this notification pops up in Home Assistant:
>Login attempt or request with invalid authentication from localhost (127.0.0.1). See the log for details.

When checking the logs, this comes from one of my iOS devices:
>Logger: homeassistant.components.http.ban
Source: components/http/ban.py:136
integration: HTTP (documentation, issues)
First occurred: November 8, 2024 at 23:20:21 (2 occurrences)
Last logged: November 8, 2024 at 23:20:22
>
>Login attempt or request with invalid authentication from localhost (127.0.0.1). Requested URL: '/api/websocket'. (Home Assistant/2024.10 (io.robbie.HomeAssistant; build:2024.934; iOS 18.1.0))

The iOS version and the TestFlight build number make it clear that it's me and not someone else trying to hack my network.

So far every time I saw the error it was pointing to `/api/websocket`

**To Reproduce**
Sadly I don't have repro steps, but it seems to affect a large number of people:
https://github.com/home-assistant/iOS/issues/2486
https://www.reddit.com/r/homeassistant/comments/1gjhlqt/login_attempt_failed/

**Expected behavior**

**Screenshots**

**Additional context**
I use Nabu Casa. No custom networking setup.

**Comment by ostgardh** [comment]
Same for me
Model Name: iPhone 15 Pro
Software Version: iOS 18.1
App version: 2024.934

Home Assistant Core Version
2024.11.1

And using nabu casa

**Comment by Nezz** [comment]
Still happening:

>2025-01-14 14:19:17.258 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from localhost (127.0.0.1). Requested URL: '/api/websocket'. (Home Assistant/2024.12.2 (io.robbie.HomeAssistant; build:2024.1058; iOS 18.2.1))
2025-01-14 14:27:57.684 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from localhost (127.0.0.1). Requested URL: '/api/websocket'. (Home Assistant/2024.12.2 (io.robbie.HomeAssistant; build:2024.1058; iOS 18.2.1))

**Comment by simon-bd** [comment]
I have the sameish thing : 
2025-01-14 10:06:08.923 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from iPhone.localdomain (10.10.10.1xx). Requested URL: '/api/template'. (Home Assistant/2024.12.3 (io.robbie.HomeAssistant; build:2024.1066; iOS 18.1.1))

**Comment by bgoncal** [comment]
Do you have widgets in the lock screen wich renders a template?

**Comment by simon-bd** [comment]
This is my code for my widget - @bgoncal :

first :
{% if is_state('switch.norma_bach_jessen_timer', 'on') %}   😴 Sleeping {% else %}   Last woke: {{ states.sensor.norma_bach_jessen_last_sleep.attributes.end | timestamp_custom('%H:%M:%S') }} {% endif %} 
second : 
{% if is_state('switch.norma_bach_jessen_timer', 'on') %}
  Started sleeping: {{ as_timestamp(states.sensor.norma_timer_since_start.last_changed) | timestamp_custom('%H:%M:%S') }}
{% else %}
  Last woke: {{ as_timestamp(states.sensor.norma_bach_jessen_last_sleep.attributes.end) | timestamp_custom('%H:%M:%S') }}
{% endif %}
third field is empty

**Comment by simon-bd** [comment]
@bgoncal what i forgot to add is that i run this on both mine and my partners phone, she is not an admin in homeassistant and for her the widget dont work and it's from her device i'm reciveing the failed login, on mine (admin) it's no problem

**Comment by simon-bd** [comment]
changed it to {{ state_attr('sensor.norma_bach_jessen_last_sleep', 'end')[11:16] }}, will submit in 15 minutes it it's still failing to display, thank you!

**Comment by simon-bd** [comment]
I've sent you the logs @bgoncal

**Comment by bgoncal** [comment]
Requested access

**Comment by bgoncal** [comment]
@simon-bd I see `Failed to render template for details widget: external(HAKit.HAError.ExternalError(code: "401", message: "401: Unauthorized"))` in her logs, besides not being admin, does she have any other restriction? Like only being able to login locally for example

**Comment by simon-bd** [comment]
No other restrictions. Using nabucasa url when not at home otherwise local ip

**Comment by bgoncal** [comment]
I don't see a direct relation to that in the logs, so perhaps, can you try removing the server from the App, force closing it and adding it back? Was it configured a long time ago?

**Comment by Nezz** [comment]
Another one:
>2025-01-17 23:09:09.322 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from localhost (127.0.0.1). Requested URL: '/api/websocket'. (Home Assistant/2025.1 (io.robbie.HomeAssistant; build:2025.1073; iOS 18.2.0))

Let me know if there is any further info I can gather to figure this out.

**Comment by simon-bd** [comment]
@bgoncal tried multiple times to uninstall reconfigure server etc. but still get the same result

**Comment by simon-bd** [comment]
From her phone logs:
2025-01-18 19:33:17.000 [Error] [com.apple.root.user-initiated-qos.cooperative] [WidgetDetailsAppIntentTimelineProvider.swift:77] entry(for:in:) > Failed to render template for details widget: external(HAKit.HAError.ExternalError(code: "401", message: "401: Unauthorized"))

From HA logs:
2025-01-18 19:33:16.973 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from localhost (127.0.0.1). Requested URL: '/api/template'. (Home Assistant/2025.1 (io.robbie.HomeAssistant; build:2025.1073; iOS 18.1.1))

**Comment by bgoncal** [comment]
@simon-bd  Can you temporaly make your wife admin to see if thats the problem? As fast as I was aware, Home Assistant didn't have user access level linked o the API

**Comment by bgoncal** [comment]
@Nezz no chance that you have the App setup on another device that is leading to this log? Like a secondary phone for example. If you update the testflight version now does it also reflect in the logs? 
Also, are you able to execute scripts from a home assistant widget for example? (it uses the websocket connection)

**Comment by simon-bd** [comment]
after i have updated her account to admin the widget is rendering, she is on the testflight version on her phone, can try to update it if you included some new logging in 2025.1.1 (2025.1007), but seems like he account was restriced when she was not an admin. can i use the same url for the logs if you need them?

**Comment by bgoncal** [comment]
@simon-bd I just checked with Core developers and indeed, templating is restricted to admin users, I had no idea, for now all I can do is to add a warning in the App, meanwhile I will take that in consideration for future implementations, there is a "widget builder" coming which will facilitate all this process.

**Comment by simon-bd** [comment]
@bgoncal Good to know, she'll have the priviliage of being admin until you are ready with more, the builder sounds amazing! thank you

**Comment by bgoncal** [comment]
@Nezz well, it shouldn't be the root cause... maybe it is the trigger but the issue must be somewhere else, can you try what I asked in the previous comment?

**Comment by Nezz** [comment]
Yes, the scripts work from iOS widgets (although I did not use them previously). The invalid auth error appears every couple of days and using widgets did not trigger it.

**Comment by Nezz** [comment]
My widgets stopped working the day after I set them up (I get a notification that it was executed, but it wasn't really). However, there are no authentication errors coming from them.

**Comment by bgoncal** [comment]
@Nezz Latest testflight build is 2025.1.1, are you on 2025.1?

**Comment by Nezz** [comment]
Yes, I switched from TestFlight to the regular version to see if it helps.

**Comment by Nezz** [comment]
The websocket one is still happening for me:
```
2025-03-07 21:44:14.261 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from localhost (127.0.0.1). Requested URL: '/api/websocket'. (Home Assistant/2025.2 (io.robbie.HomeAssistant; build:2025.1152; iOS 18.3.1))
2025-03-07 21:44:16.304 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from localhost (127.0.0.1). Requested URL: '/api/websocket'. (Home Assistant/2025.2 (io.robbie.HomeAssistant; build:2025.1152; iOS 18.3.1))
2025-03-07 21:44:27.581 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from localhost (127.0.0.1). Requested URL: '/api/websocket'. (Home Assistant/2025.2 (io.robbie.HomeAssistant; build:2025.1152; iOS 18.3.1))
```

**Comment by techniker** [comment]
Same issue here with the stock iOS HomeAssistant App.
If a user is not an Admin, his App frequently generates Login attempt failed Notifications.

**Comment by ajuch** [comment]
Is there any progress? I cannot get this to work for some time. I increased the max failed login attempts to 10, re-installed the IOS app, deleted all safari cookies, but still it's repeatedly trying to authenticate with wrong credentials directly after entering the server address.

```
Nov 09 21:52:22 hass homeassistant[1227775]: 2025-11-09 21:52:22.070 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.30.100.147 (172.30.100.147). Requested URL: '/auth/token'. (Home Assistant/2025.10.0 (io.robbie.HomeAssistant; build:2025.1443; iOS 18.6.2) Alamofire/5.8.0)
Nov 09 21:52:22 hass homeassistant[1227775]: 2025-11-09 21:52:22.109 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.30.100.147 (172.30.100.147). Requested URL: '/auth/token'. (Home Assistant/2025.10.0 (io.robbie.HomeAssistant; build:2025.1443; iOS 18.6.2) Alamofire/5.8.0)
Nov 09 21:52:22 hass homeassistant[1227775]: 2025-11-09 21:52:22.147 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.30.100.147 (172.30.100.147). Requested URL: '/auth/token'. (Home Assistant/2025.10.0 (io.robbie.HomeAssistant; build:2025.1443; iOS 18.6.2) Alamofire/5.8.0)
Nov 09 21:52:22 hass homeassistant[1227775]: 2025-11-09 21:52:22.182 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.30.100.147 (172.30.100.147). Requested URL: '/auth/token'. (Home Assistant/2025.10.0 (io.robbie.HomeAssistant; build:2025.1443; iOS 18.6.2) Alamofire/5.8.0)
Nov 09 21:52:22 hass homeassistant[1227775]: 2025-11-09 21:52:22.215 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.30.100.147 (172.30.100.147). Requested URL: '/auth/token'. (Home Assistant/2025.10.0 (io.robbie.HomeAssistant; build:2025.1443; iOS 18.6.2) Alamofire/5.8.0)
Nov 09 21:52:22 hass homeassistant[1227775]: 2025-11-09 21:52:22.248 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.30.100.147 (172.30.100.147). Requested URL: '/auth/token'. (Home Assistant/2025.10.0 (io.robbie.HomeAssistant; build:2025.1443; iOS 18.6.2) Alamofire/5.8.0)
Nov 09 21:52:22 hass homeassistant[1227775]: 2025-11-09 21:52:22.276 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.30.100.147 (172.30.100.147). Requested URL: '/auth/token'. (Home Assistant/2025.10.0 (io.robbie.HomeAssistant; build:2025.1443; iOS 18.6.2) Alamofire/5.8.0)
Nov 09 21:52:22 hass homeassistant[1227775]: 2025-11-09 21:52:22.304 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.30.100.147 (172.30.100.147). Requested URL: '/auth/token'. (Home Assistant/2025.10.0 (io.robbie.HomeAssistant; build:2025.1443; iOS 18.6.2) Alamofire/5.8.0)
Nov 09 21:52:22 hass homeassistant[1227775]: 2025-11-09 21:52:22.340 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.30.100.147 (172.30.100.147). Requested URL: '/auth/token'. (Home Assistant/2025.10.0 (io.robbie.HomeAssistant; build:2025.1443; iOS 18.6.2) Alamofire/5.8.0)
Nov 09 21:52:22 hass homeassistant[1227775]: 2025-11-09 21:52:22.366 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.30.100.147 (172.30.100.147). Requested URL: '/auth/token'. (Home Assistant/2025.10.0 (io.robbie.HomeAssistant; build:2025.1443; iOS 18.6.2) Alamofire/5.8.0)
Nov 09 21:52:22 hass homeassistant[1227775]: 2025-11-09 21:52:22.366 WARNING (MainThread) [homeassistant.components.http.ban] Banned IP 172.30.100.147 for too many login attempts
```

I have migrated the home-assistant server some time ago, but I think deleting the app should remove any old authentication data on the device. If not, please give me some advice!

**Comment by bgoncal** [comment]
@ajuch Your server information is saved on iOS Keychain for security reasons, since Apple does not allow developers to know when an App is deleted, we can't remove those, so when you reinstall they will still be there.

Open companion App settings and remove the server manually.

**Comment by ajuch** [comment]
Thanks, @bgoncal I removed the server, didn't realise it was still there. After unblocking and restarting Home-Assistant, I could log in successfully, but immediately got blocked again:

```
Nov 10 21:28:01 hass homeassistant[1341608]: 2025-11-10 21:28:01.475 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.30.100.147 (172.30.100.147). Requested URL: '/auth/token'. (Home Assistant/2
025.10.0 (io.robbie.HomeAssistant; build:2025.1443; iOS 18.6.2) Alamofire/5.8.0)
Nov 10 21:28:01 hass homeassistant[1341608]: 2025-11-10 21:28:01.484 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.30.100.147 (172.30.100.147). Requested URL: '/api/websocket'. (Mozilla/5.0 (
iPad; CPU OS 18_6_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Home Assistant/2025.10.0 (io.robbie.HomeAssistant; build:2025.1443; iPadOS 18.6.2) Mobile/HomeAssistant, like Safari)
Nov 10 21:28:01 hass homeassistant[1341608]: 2025-11-10 21:28:01.491 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from 172.30.100.147 (172.30.100.147). Requested URL: '/auth/token'. (Home Assistant/2
025.10.0 (io.robbie.HomeAssistant; build:2025.1443; iOS 18.6.2) Alamofire/5.8.0)
Nov 10 21:28:01 hass homeassistant[1341608]: 2025-11-10 21:28:01.492 WARNING (MainThread) [homeassistant.components.http.ban] Banned IP 172.30.100.147 for too many login attempts
```
After removing the entry in the ip_bans file and restarting Home-Assistant, now the app is working!

**Comment by IvoKvesic** [comment]
See also my report related to the same issue at: https://github.com/home-assistant/core/issues/181148. Home Assistant currently provides no way to exempt a known local IP or subnet from ip_ban while retaining brute-force protection for untrusted sources. A per-IP/CIDR ip_ban exemption would substantially mitigate this failure mode.
